src/test/java/BoardTest.java

```diff
@@ -243,4 +243,131 @@ public void testMovePawnsIllegally() throws InvalidPositionException
 		/*16 b/c there are 16 pawns that get moved illegally*/
 		assertEquals(numErrors, 16);
 	}
+	
+	//Moves pawns forward to test functionality of moving pieces to valid locations. 
+	@Test
+	public void testValidMoveToNewPosition() 
+	{
+		Board board = new Board();
+		try
+		{
+			board.movePiece(new Position(1,0), new Position(3, 0));
+			board.movePiece(new Position(1,1), new Position(3,1));
+			board.movePiece(new Position(1,4), new Position(3,4));
+			board.movePiece(new Position(0,0), new Position(2,0));
+			board.movePiece(new Position(0,1), new Position(2,2));
+			board.movePiece(new Position(0,2), new Position(1,1));
+			board.movePiece(new Position(0,3), new Position(1,4)); 
+			board.movePiece(new Position(0,4), new Position(0,3));
+			
+		}
+		catch(InvalidPositionException ex)
+		{
+			fail(ex.getLocalizedMessage());
+		}
+	}
+	
+	//Error check for pieces moved off the board. 
+	@Test 
+	public void testInvalidMoveToOutsideGrid()
+	{
+		Board board = new Board();
+		int numErrors = 0;
+		for(int catchNum = 0; catchNum < 12; catchNum++){
+			try
+			{
+				int row = 0;
+				for(int col = 0; col < 8; col++)
+				{
+					board.movePiece(new Position(row, col), new Position(row, col-1));
+				}
+				board.movePiece(new Position(0,0), new Position(-1,0));
+				board.movePiece(new Position(1,0), new Position(-2,0));
+				board.movePiece(new Position(0,7), new Position(0,8));
+				board.movePiece(new Position(1,7), new Position(1,8));
+			}
+			catch(InvalidPositionException ex)
+			{
+				numErrors += 1;
+			}
+		}
+		/*12 errors for out of bounds 
+		  8 to check the first row moving backwards
+		  4 for checking corners */
+		assertEquals(numErrors, 12);
+	}
+	
```

Comment:
- [ ] I believe these three (3) "invalid" tests are still not testing the right thing. They loop x number of times, and count the number of errors, which should equal x. But only the first `movePiece()` call in the try/catch block is getting tested. The reason the tests still pass is that that first `movePiece()` call is being called x times. If I didn't explain that well, feel free to give me a call.